packages/pipeline-editor/src/PipelineService.tsx

```diff
@@ -124,4 +127,51 @@ export class PipelineService {
       buttons: [Dialog.okButton()]
     });
   }
+
+  static convertPipeline(pipelineDefinition: any): any {
+    const pipelineJSON = JSON.parse(JSON.stringify(pipelineDefinition));
+
+    const currentVersion: number =
+      +Utils.hasPipelineAppdataField(pipelineJSON.pipelines[0], 'version') || 0;
+    if (currentVersion < 1) {
+      // original pipeline definition without a version
+      console.log('Migrating pipeline to version 1');
```

Comment:
We should move this outside the set of if statements to avoid multiple log statements and just provide the destination (or simply 'current version').
```suggestion
 console.log('Migrating pipeline to the current version.');
```
If this is more for debug and you _do_ want this logged on each step, then the log statement should be moved into the respective step's method.